advanced_functionality/causal-inference/container/causal_nex/train

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+
+# A sample training component that trains a simple scikit-learn decision tree model.
+# This implementation works in File mode and makes no assumptions about the input file names.
+# Input is specified as CSV with a data point in each row and the labels in the first column.
+
+from __future__ import print_function
+
+import json
+import os
+import pickle
+import sys
+import traceback
+
+import pandas as pd
+from causalnex.discretiser import Discretiser
+import warnings
+from causalnex.structure import StructureModel
+from sklearn.model_selection import train_test_split
+from causalnex.network import BayesianNetwork
+from causalnex.evaluation import classification_report
+from causalnex.evaluation import roc_auc
+
+# These are the paths to where SageMaker mounts interesting things in your container.
+prefix = '/opt/ml/'
+
+input_path = prefix + 'input/data'
+output_path = os.path.join(prefix, 'output')
+model_path = os.path.join(prefix, 'model')
+param_path = os.path.join(prefix, 'input/config/hyperparameters.json')
+
+# This algorithm has a single channel of input data called 'training'. Since we run in
+# File mode, the input files are copied to the directory specified here.
+channel_name='training'
+training_path = os.path.join(input_path, channel_name)
+
+# The function to execute the training.
+def train():
+    print('Starting the training.')
+    try:
+        # Take the set of files and read them all into a single pandas dataframe
+        input_files = [ os.path.join(training_path, file) for file in os.listdir(training_path) ]
+        if len(input_files) == 0:
+            raise ValueError(('There are no files in {}.\n' +
+                              'This usually indicates that the channel ({}) was incorrectly specified,\n' +
+                              'the data specification in S3 was incorrectly specified or the role specified\n' +
+                              'does not have permission to access the data.').format(training_path, channel_name))
+        
+        raw_data =[]
+        for file in input_files:
+            if '.csv' in file:
+                raw_data.append(pd.read_csv(file, encoding='utf8', engine='python'))
+        train_data = pd.concat(raw_data)
+
+        # Discretize the data with causalnex functions
```

Comment:
Clarity nit: what are the magic numbers chosen for `numeric_split_points` values in lines 56-61?